pkg/api/roles.go

```diff
@@ -50,157 +50,189 @@ var (
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionProvisioningReload,
+					Scope:  ScopeProvisionersAll,
 				},
 			},
-			Grants: []string{accesscontrol.RoleGrafanaAdmin},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:datasources:admin",
-				Description: "Gives access to create, read, update, delete datasources",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionDatasourcesRead,
-						Scope:  ScopeDatasourcesAll,
-					},
-					{
-						Action: ActionDatasourcesWrite,
-						Scope:  ScopeDatasourcesAll,
-					},
-					{
-						Action: ActionDatasourcesCreate,
-					},
-					{
-						Action: ActionDatasourcesDelete,
-						Scope:  ScopeDatasourcesAll,
-					},
-					{
-						Action: ActionDatasourcesQuery,
-						Scope:  ScopeDatasourcesAll,
-					},
+		Grants: []string{accesscontrol.RoleGrafanaAdmin},
+	}
+
+	datasourcesReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:datasources:reader",
+			DisplayName: "Data source reader",
+			Description: "Read and query all datasources.",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionDatasourcesRead,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionDatasourcesQuery,
+					Scope:  ScopeDatasourcesAll,
 				},
 			},
-			Grants: []string{string(models.ROLE_ADMIN)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     2,
-				Name:        "fixed:datasources:id:viewer",
-				Description: "Gives access to read datasources ID",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionDatasourcesIDRead,
-						Scope:  ScopeDatasourcesAll,
-					},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	datasourcesWriterRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:datasources:writer",
+			DisplayName: "Data source writer",
+			Description: "Create, update, delete, read, or query data sources.",
+			Permissions: accesscontrol.ConcatPermissions(datasourcesReaderRole.Role.Permissions, []accesscontrol.Permission{
+				{
+					Action: ActionDatasourcesWrite,
+					Scope:  ScopeDatasourcesAll,
 				},
-			},
-			Grants: []string{string(models.ROLE_VIEWER)},
+				{
+					Action: ActionDatasourcesCreate,
+				},
+				{
+					Action: ActionDatasourcesDelete,
+					Scope:  ScopeDatasourcesAll,
+				},
+			}),
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:datasources:compatibility:querier",
-				Description: "Query data sources when data source permissions are not in use",
-				Permissions: []accesscontrol.Permission{
-					{Action: ActionDatasourcesQuery},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	datasourcesIdReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     3,
+			Name:        "fixed:datasources:id:reader",
```

Comment:
Should this be with a `.` ?
```suggestion
			Name:        "fixed:datasources.id:reader",
```